ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1021,6 +1042,39 @@ - (ORKStepViewController *)viewControllerForStep:(ORKStep *)step {
         
         stepViewController.restorationIdentifier = step.identifier;
         stepViewController.restorationClass = stepViewControllerClass;
+        if ([stepViewController isKindOfClass:[ORKReviewStepViewController class]]) {
+            ORKReviewStepViewController *reviewStepViewController = (ORKReviewStepViewController *) stepViewController;
+            NSMutableArray *steps = [[NSMutableArray alloc] init];
+            if (reviewStepViewController.reviewStep.isStandalone) {
+                for (ORKStep *step in reviewStepViewController.reviewStep.steps) {
+                    if ([self shouldReviewStep:step]) {
+                        [steps addObject:step];
+                    }
+                }
+            } else {
+                ORKStep *nextStep = step;
+                do {
+                    switch (reviewStepViewController.reviewStep.reviewDirection) {
```

Comment:
Can you use `_managedStepIdentifiers` to gather the steps? instead of using  `stepAfterStep` and `stepBeforeStep`.